src/Actor_TEST.cc

```diff
@@ -19,6 +19,127 @@
+  return (_anim1.Name() == _anim2.Name()) &&
+    (_anim1.Filename() == _anim2.Filename()) &&
+    (_anim1.FilePath() == _anim2.FilePath()) &&
+    (std::abs(_anim1.Scale() - _anim2.Scale()) < EPS) &&
+    (_anim1.InterpolateX() == _anim2.InterpolateX());
+}
+
+/////////////////////////////////////////////////
+sdf::Trajectory CreateDummyTrajectory()
+{
+  sdf::Trajectory trajectory;
+  sdf::Waypoint waypoint;
+  waypoint.SetTime(0.12);
+  waypoint.SetPose({3, 2, 1, 0, IGN_PI, 0});
+  trajectory.SetId(1234);
+  trajectory.SetType("trajectory_type");
+  trajectory.SetTension(4.567);
+  trajectory.AddWaypoint(waypoint);
+  return trajectory;
+}
+
+/////////////////////////////////////////////////
+bool TrajectoriesEqual(const sdf::Trajectory &_traj1,
+    const sdf::Trajectory &_traj2)
+{
+  constexpr double EPS = 1e-6;
+  bool waypoints_equal = true;
+  if (_traj1.WaypointCount() != _traj2.WaypointCount())
+  {
+    return false;
+  }
+  for (uint64_t wp_idx = 0; wp_idx < _traj1.WaypointCount(); ++wp_idx)
+  {
+    auto wp1 = _traj1.WaypointByIndex(wp_idx);
+    auto wp2 = _traj2.WaypointByIndex(wp_idx);
+    waypoints_equal &= (std::abs(wp1->Time() - wp2->Time()) < EPS) &&
+      wp1->Pose() == wp2->Pose();
+  }
+  return waypoints_equal &&
+    (_traj1.Id() == _traj2.Id()) &&
+    (_traj1.Type() == _traj2.Type()) &&
+    (std::abs(_traj1.Tension() - _traj2.Tension()) < EPS);
+}
+
+/////////////////////////////////////////////////
+sdf::Actor CreateDummyActor()
+{
+  sdf::Actor actor;
+  actor.SetName("test_dummy_actor");
+  actor.SetPose({3, 2, 1, 0, IGN_PI, 0});
+  actor.SetPoseFrame("ground_plane");
+  actor.SetSkinFilename("walk.dae");
+  actor.SetSkinScale(2.0);
+  actor.SetScriptLoop(true);
+  actor.SetScriptDelayStart(2.8);
+  actor.SetScriptAutoStart(false);
+  actor.SetFilePath("/home/path/to/model.sdf");
+  // Add a dummy trajectory and animation as well
+  actor.AddTrajectory(CreateDummyTrajectory());
+  actor.AddAnimation(CreateDummyAnimation());
+  return actor;
+}
+
+/////////////////////////////////////////////////
+bool ActorsEqual(const sdf::Actor &_actor1, const sdf::Actor &_actor2)
+{
+  constexpr double EPS = 1e-6;
+  // Check animations, trajectories and properties
+  bool animations_equal = true;
+  if (_actor1.AnimationCount() != _actor2.AnimationCount())
+  {
+    return false;
+  }
+  for (uint64_t anim_idx = 0; anim_idx < _actor1.AnimationCount(); ++anim_idx)
+  {
+    animations_equal &= AnimationsEqual(
+        *_actor1.AnimationByIndex(anim_idx),
+        *_actor2.AnimationByIndex(anim_idx));
+  }
+  bool trajectories_equal = true;
+  if (_actor1.TrajectoryCount() != _actor2.TrajectoryCount())
+  {
+    return false;
+  }
+  for (uint64_t traj_idx = 0; traj_idx < _actor1.TrajectoryCount(); ++traj_idx)
+  {
+    trajectories_equal &= TrajectoriesEqual(
+        *_actor1.TrajectoryByIndex(traj_idx),
+        *_actor2.TrajectoryByIndex(traj_idx));
+  }
+  return animations_equal && trajectories_equal &&
```

Comment:
nitpick: camelCase for `animations_equal` and `trajectories_equal`

Reply:
[Done](https://github.com/osrf/sdformat/pull/301/commits/f00cf338e2131fdedfe35184c14f47e36bec93d9), a quick search for `_equal` showed that `waypoints_equal` had the same issue so fixed that as well.